charts/redpanda/values.go

```diff
@@ -484,9 +484,10 @@ type PostInstallJob struct {
 	Affinity  map[string]any `json:"affinity"`
 
 	// Fields that are in values.yaml but not in values.schema.json.
-	// Enabled     bool              `json:"enabled"`
-	// Labels      map[string]string `json:"labels"`
-	// Annotations map[string]string `json:"annotations"`
+	Enabled         bool                   `json:"enabled"`
+	Labels          map[string]string      `json:"labels"`
+	Annotations     map[string]string      `json:"annotations"`
+	SecurityContext corev1.SecurityContext `json:"securityContext"`
```

Comment:
> do we have a way to audit the use of these potentially vestigial fields?

No, there is no way to check what values are used in the wild.

> which this PR currently doesn't fix - should it?

In the new commit that would have dedicated values input to test it out it can go in this PR or in new. I'm open to any outcome. 

>  I'm also not 100% convinced that additional security context settings can provide anything that this script could conceivably require in order to operate (outside of pathological scenarios).

I feel like access to `/tmp/` file system and network connectivity should be an requirement. Other security settings could be more restrictive.